vimiv/imutils/exif.py

```diff
@@ -261,13 +293,95 @@ def copy_exif(self, dest: str, reset_orientation: bool = True) -> None:
         except OSError as e:
             _logger.debug("Failed to write exif data for '%s': '%s'", dest, str(e))
 
-    def exif_date_time(self) -> str:
+    def get_date_time(self) -> str:
         with contextlib.suppress(KeyError):
             return self._metadata["Exif.Image.DateTime"].raw_value
         return ""
 
 
-has_exif_support = ExifHandler != _ExifHandlerBase
+has_exif_support = _ExternalKeyHandler != _ExternalKeyHandlerBase
+
+
+class MetadataHandler:
+    """Handler to load and copy exif information of a single image.
+
+    This class provides the interface for handling exif support. By default none of the
+    operations are implemented. Instead it is up to a child class which wraps around one
+    of the supported exif libraries to implement the methods it can.
+    """
+
+    def __init__(self, filename=""):
+        self.filename = filename
+        self._internal_handler: Optional[_InternalKeyHandler] = None
+        self._external_handler: Optional[_ExternalKeyHandler] = None
+
+    @property
+    def internal_handler(self) -> _InternalKeyHandler:
+        if self._internal_handler is None:
+            self._internal_handler = _InternalKeyHandler(self.filename)
+        return self._internal_handler
+
+    @property
+    def external_handler(self) -> _ExternalKeyHandler:
+        if self._external_handler is None:
+            self._external_handler = _ExternalKeyHandler(self.filename)
+        return self._external_handler
+
+    def copy_metadata(self, dest: str, reset_orientation: bool = True) -> None:
+        """Copy exif information from current image to dest.
+
+        Args:
+            dest: Path to write the exif information to.
+            reset_orientation: If true, reset the exif orientation tag to normal.
+        """
+        try:
+            self.external_handler.copy(dest, reset_orientation)
+        except UnsupportedExifOperation:
+            # Todo
+            pass
+
+    def get_date_time(self) -> str:
+        """Get exif creation date and time as formatted string."""
+        try:
+            self.external_handler.get_data_time()
+        except UnsupportedExifOperation:
+            # Todo
+            pass
+
+    def get_formatted_metadata(
+        self, desired_keys: Sequence[str]
+    ) -> Dict[Any, Tuple[str, str]]:
+        metadata = dict()
+
+        for base_key in desired_keys:
+            base_key = base_key.strip()
+
+            try:
+                key, key_name, key_value = self._fetch_key(base_key)
+                metadata[key] = key_name, key_value
+            except (KeyError, TypeError):
```

Comment:
How about also catching the `UnsupportedExifOperation` here for completeness, instead of handling it in `_fetch_keys` directly? We will probably never use the class without exif support, but safety first, you are right :blush:

Reply:
As with the `get_keys` methods, I think it makes sense to propagate the excepion here too since this method can be called by cliets via the API anytime.